src/redux/greetings/greetingsSlice.js

```diff
@@ -0,0 +1,33 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = {
+  greeting: '',
+  error: null,
+};
+
+export const getRandomGreeting = createAsyncThunk('getRandomGreeting', async () => {
+  try {
+    const response = await axios.get('http://localhost:3000/api/greeting');
```

Comment:
- [ ] Your frontend does not fetch the greeting from the API due to cross-origin blockage so I cannot see nay messages. In your backend, you need to install the `rack-cors` gem and configure the middleware to allow requests from the frontend.

  See this link for setting up CORS in the backend: https://github.com/cyu/rack-cors

  ![image](https://github.com/mrkamin/hello-react-front-end/assets/81468675/77bb025a-75ba-483a-9b29-22bf191b44c7)

Reply:
@julie-ify,
thanks for your good and valuable feedback, 
the cors setup according to the link you provided.